include/fp_util/feature_detect.h

```diff
@@ -0,0 +1,18 @@
+/**
+ * Author:   Lukas
+ * Created:  18/11/2023.
+ **/
+
+#ifndef FEATURE_DETECT_H
+#define FEATURE_DETECT_H
+
+namespace fp_util {
+
+__attribute__((feature_variable("__VARA__DETECT__"))) void detect() {
+  long counter = 0;
```

Comment:
If you want to count them, this needs to be static, when used in a header even static inline, which might not work for all cases, as static inline is a newer C++ feature.

Reply:
For my cases I don't really need the counter variable, this is just a copy/paste from the busy sleep.

I just need to ensure that whatever happens inside the function does not get optimized away by the compiler and wasn't sure exactly how much I need for that.
 
Renaming the variable might be better to avoid any confusion.